bpfd/src/bpf.rs

```diff
@@ -913,31 +931,33 @@ impl BpfManager {
                 if let Some(map) = self.maps.get_mut(&m) {
                     map.used_by.push(id);
 
-                    // If map owner program still exists update it's maps_used_by_field
-                    if let Some(program) = self.programs.get_mut(&m) {
-                        program
-                            .data_mut()?
-                            .set_maps_used_by(Some(map.used_by.clone()));
+                    // Update all the programs using the same map with the updated map_used_by.
+                    for used_by_id in map.used_by.iter() {
+                        if let Some(program) = self.programs.get_mut(used_by_id) {
+                            if let Ok(data) = program.data_mut() {
+                                data.set_maps_used_by(Some(map.used_by.clone()));
+                            }
+                        }
                     }
                 } else {
                     return Err(BpfdError::Error("map_owner_id does not exists".to_string()));
                 }
             }
             None => {
-                let program = self
-                    .programs
-                    .get_mut(&id)
-                    .expect("Program should be loaded");
+                if let Some(program) = self.programs.get_mut(&id) {
+                    let map = BpfMap { used_by: vec![id] };
 
-                let map = BpfMap { used_by: vec![id] };
+                    self.maps.insert(id, map);
 
-                self.maps.insert(id, map);
-
-                // TODO(astoycos) remove external self.maps, keep all map tracking info in Program
-                // update programs map_used_by
-                program.data_mut()?.set_maps_used_by(Some(vec![id]));
+                    // Update this program with the updated map_used_by
+                    if let Ok(data) = program.data_mut() {
+                        data.set_maps_used_by(Some(vec![id]));
+                    }
+                }
 
-                set_dir_permissions(map_pin_path.to_str().unwrap(), MAPS_MODE).await;
+                if let Some(path) = map_pin_path.to_str() {
```

Comment:
Wait we should panic if we can't convert the map_pin_path to a string right? (i.e that shouldn't happen)

Reply:
I can hear the argument for not panicing but with this we're doing nothing if the `to_str()` fails